src/vue/forms/AssetUpdateForm.vue

```diff
@@ -0,0 +1,365 @@
+    isLoaded: false,
+    isLoadingFailed: false,
+    currentStep: 1,
+    STEPS,
+    DOCUMENT_TYPES,
+    ASSET_POLICIES,
+    NAME_MAX_LENGTH,
+  }),
+
+  validations () {
+    return {
+      form: {
+        information: {
+          name: {
+            required,
+            maxLength: maxLength(NAME_MAX_LENGTH),
+          },
+        },
+      },
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      account: vuexTypes.account,
+    }),
+
+    assetRequestOpts () {
+      const requestId = this.request.id || ASSET_CREATION_REQUEST_ID
+      const logo = this.form.information.logo
+      const terms = this.form.advanced.terms
+
+      return {
+        requestID: requestId,
+        code: this.assetForUpdate,
+        policies: this.form.information.policies.reduce((a, b) => (a | b), 0),
+        details: {
+          name: this.form.information.name,
+          logo: logo ? logo.getDetailsForSave() : EMPTY_DOCUMENT,
+          terms: terms ? terms.getDetailsForSave() : EMPTY_DOCUMENT,
+        },
+      }
+    },
+  },
+
+  async created () {
+    try {
+      this.request = await this.getAssetRequestForUpdate(this.assetForUpdate)
+      this.populateForm()
+      this.isLoaded = true
+    } catch (e) {
+      this.isLoadingFailed = true
+      ErrorHandler.processWithoutFeedback(e)
+    }
+  },
+
+  methods: {
+    populateForm () {
+      this.form = {
+        information: {
+          name: this.request.assetName,
+          logo: this.request.logo.key
+            ? new DocumentContainer(this.request.logo)
+            : null,
+          policies: this.request.policies,
+        },
+        advanced: {
+          terms: this.request.terms.key
+            ? new DocumentContainer(this.request.terms)
+            : null,
+        },
+      }
+    },
+
+    async getAssetRequestForUpdate (assetCode) {
+      try {
+        const { data } = await Sdk.horizon.request.getAllForAssets({
+          requestor: this.account.accountId,
+          asset: assetCode,
+        })
+        const requests = data.map(request => RecordWrapper.request(request))
+
+        const assetUpdateRequests =
+          requests.filter(r => r instanceof AssetUpdateRequestRecord)
+
+        return this.getLatestUpdateRequest(assetUpdateRequests)
+      } catch (e) {
+        ErrorHandler.processWithoutFeedback(e)
+      }
+    },
+
+    getLatestUpdateRequest (assetUpdateRequests) {
+      const latestApprovedRequestTime = moment.max(
+        assetUpdateRequests
+          .filter(request => request.isApproved)
+          .map(request => moment(request.updatedAt))
+      )
+
+      const latestApprovedRequest = assetUpdateRequests.find(request => {
+        return request.isApproved &&
+          latestApprovedRequestTime.isSame(request.updatedAt)
+      })
+
+      const latestUpdatableRequest = assetUpdateRequests
+        .find(request => request.isPending || request.isRejected)
+
+      return latestUpdatableRequest || latestApprovedRequest
```

Comment:
you can't update approved the request, then why are you returning it here?

Reply:
Removed approved request, used asset record.